lib/geordi/dump_loader.rb

```diff
@@ -6,24 +6,43 @@
 module Geordi
   class DumpLoader
 
-    def initialize(file)
+    def initialize(file, database)
       @dump_file = file
+      @database = database
     end
 
     def development_database_config
+      return @config if @config
+
       require 'yaml'
 
       evaluated_config_file = ERB.new(File.read('config/database.yml')).result
 
       # Allow aliases and a special set of classes like symbols and time objects
       permitted_classes = [Symbol, Time]
-      @config ||= if Gem::Version.new(Psych::VERSION) >= Gem::Version.new('3.1.0')
+      database_config = if Gem::Version.new(Psych::VERSION) >= Gem::Version.new('3.1.0')
         YAML.safe_load(evaluated_config_file, permitted_classes: permitted_classes, aliases: true)
       else
         YAML.safe_load(evaluated_config_file, permitted_classes, [], true)
       end
 
-      @config['development']
+      development_config = database_config['development']
+
+      if development_config.values[0].is_a? Hash # Multi-db setup
+        if @database
+          @config = development_config[@database] || Interaction.fail(%(Unknown development database "#{@database}".))
+        else
+          Interaction.fail 'Multiple databases detected. Select the database to load into with `-L database`'
```

Comment:
When working with multiple databases, the developer either is fully aware of it (because he is working with all databases alike) and thus will always pass a database name when sourcing a dump (the `if` case above). If there is one "main" database and one or more "additional" databases, the developer will usually not think about "multiple databases" and we do not want to force him to, either. 

Please re-align this to the behavior of `dumple`: fall back to the "primary" database, and default to the first entry.

@foobear what do you think?

Reply:
I think it makes sense to use the `primary` or first database, if no database was specified (like `geordi dump` does).